kytos/core/config.py

```diff
@@ -125,18 +130,55 @@ def parse_args(self):
         options, argv = self.conf_parser.parse_known_args()
 
         config = ConfigParser()
-        result = config.read([options.conf or defaults.get('conf')])
 
-        if result:
-            defaults.update(dict(config.items("daemon")))
-        else:
-            print('There is no config file.')
-            exit(-1)
+        config_file = options.conf or defaults.get('conf')
+
+        if not os.path.exists(config_file):
+            self._generate_file_from_template(TEMPLATE_FILES, BASE_ENV,
+                                              prefix=BASE_ENV,
+                                              syslog_args=SYSLOG_ARGS)
+
+        config.read(config_file)
+
+        defaults.update(dict(config.items("daemon")))
 
         self.parser.set_defaults(**defaults)
 
         self.options['daemon'] = self._parse_options(argv)
 
+    @staticmethod
+    def _generate_file_from_template(templates,
+                                     destination=Path(__file__).parent,
+                                     **kwargs):
+        """Create a config file based on a template file.
+
+        If no destination is passed, the new conf file will be created on the
+        directory of the template file.
+
+        Args:
+            template (string):    Path of the template file
+            destination (string): Directory in which the config file will
+                                  be placed.
+        """
+        from jinja2 import Template
+
+        if str(kwargs['prefix']) != '/':
+            kwargs['prefix'] = Path(str(kwargs['prefix']).rstrip('/'))
+        kwargs['jwt_secret'] = uuid.uuid4().hex
+
+        # Create the paths used by Kytos.
+        directories = [os.path.join(BASE_ENV, 'etc/kytos')]
+        for directory in directories:
+            if not os.path.exists(directory):
+                os.makedirs(directory)
+
+        for path in templates:
+            with open(path, 'r', encoding='utf-8') as src_file:
```

Comment:
Running this modification, the error FileNotFound is displayed.

The error message:

FileNotFoundError: [Errno 2] No such file or directory: 'etc/kytos/kytos.conf.template'

To reproduce this error:

- Create a new virtual environment.
- Clone this repository.
- Get this modification: git fetch origin refs / pull / 1044 / head
- make checkout for this new branch: git checkout -b josemauro / fix-issue-1042 FETCH_HEAD
- Install kytos.
- Run kytos: kytosd -f